service/frontend/workflow_handler.go

```diff
@@ -1247,14 +1250,17 @@ func (wh *WorkflowHandler) RecordActivityTaskHeartbeatById(ctx context.Context,
 		return nil, errActivityIDNotSet
 	}
 
-	taskToken := &tokenspb.Task{
-		NamespaceId:      namespaceID.String(),
-		RunId:            runID,
-		WorkflowId:       workflowID,
-		ScheduledEventId: common.EmptyEventID,
-		ActivityId:       activityID,
-		Attempt:          1,
-	}
+	taskToken := tasktoken.NewActivityTaskToken(
+		namespaceID.String(),
+		workflowID,
+		runID,
+		common.EmptyEventID,
+		activityID,
+		"",
+		1,
+		nil,
+		common.EmptyVersion,
```

Comment:
any future plan to use a real version?

Reply:
no, the *ActivitybyID APIs are on the deprecation path, and the contract does not include clock or version